src/components/CheckIn/AddPieceModal.js

```diff
@@ -47,20 +52,68 @@ const footer = (close, save, checkIn) => (
   </ModalFooter>
 );
 
+const FIELD_ATTRS_FOR_REQUIRED_LOCATION = {
+  required: true,
+  validate: Required,
+};
+
 class AddPieceModal extends Component {
   static propTypes = {
+    change: PropTypes.func.isRequired,
     checkIn: PropTypes.func.isRequired,
     close: PropTypes.func.isRequired,
+    createInventoryValues: PropTypes.object.isRequired,
+    dispatch: PropTypes.func.isRequired,
     handleSubmit: PropTypes.func.isRequired,
     locations: PropTypes.arrayOf(PropTypes.shape({
       label: PropTypes.string.isRequired,
       value: PropTypes.string.isRequired,
     })),
     showPieceFormatField: PropTypes.bool,
     store: PropTypes.object.isRequired,
+    submitting: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAddItemButtonDisabled: true,
+      validateLocation: {},
+    };
   }
 
-  checkIn = () => this.props.checkIn(getFormValues('AddPieceModalForm')(this.props.store.getState()));
+  componentDidMount() {
+    this.validateRequiredFields();
+  }
+
+  checkIn = () => this.props.checkIn(getFormValues(ADD_PIECE_MODAL_FORM)(this.props.store.getState()));
+
+  onChangeSelect = (_, value, prevValue, fieldName) => {
+    const { dispatch, change } = this.props;
+
+    dispatch(change(fieldName, value));
```

Comment:
Do we really need to dispatch change here? redux form guide says about if we don't call event.preventDefault new value will put into store https://redux-form.com/7.1.0/docs/api/field.md/

Reply:
I partially agree with Anddrews. But my opinion - we don't need this handler at all.

Reply:
Fixed